packages/core/src/external/fhir/patient/reference.ts

```diff
@@ -0,0 +1,682 @@
+import { Patient, Reference, Resource } from "@medplum/fhirtypes";
+import { isReferenceOfType, toReference } from "../shared/references";
+
+/**
+ * Returns the Patient references from the given resources.
+ *
+ * @param resources Array of Resource to search for Patient references.
+ * @returns Array of Reference<Patient>
+ * @see https://www.hl7.org/fhir/r4/patient.html ("This resource is referenced by...")
+ */
+export function getPatientReferencesFromResources(resources: Resource[]): Reference<Patient>[] {
+  const references: Reference<Patient>[] = [];
+  for (const resource of resources) {
+    if (resource.resourceType === "Patient") {
```

Comment:
isnt this already done in `getPatientIdsFromPatient`? I suggest we leave it here and remove the call to `getPatientIdsFromPatient`, since it will be one less iteration through the set of all resources

Reply:
Now that you brought it up I think this is incorrect - we're creating a reference that might not exist to represent a resource.

I'll update it removing this `if` and leave the `getPatientIdsFromPatient`.

IMO we should not build functions expecting the behavior of other functions to be one way or another, and having them separated is clearer, and easier to test and maintain.